cmd/finch/version_test.go

```diff
@@ -5,44 +5,127 @@ package main
 
 import (
 	"bytes"
-	"io"
-	"os"
 	"testing"
 
+	"github.com/spf13/cobra"
+
+	"github.com/runfinch/finch/pkg/mocks"
+
+	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
-func captureStdout(t *testing.T, f func()) string {
-	temp := os.Stdout
-	read, write, _ := os.Pipe()
-	os.Stdout = write
+func TestVersionCommand(t *testing.T) {
+	t.Parallel()
 
-	f()
+	ctrl := gomock.NewController(t)
+	logger := mocks.NewLogger(ctrl)
+	lcc := mocks.NewLimaCmdCreator(ctrl)
+	var stdOut bytes.Buffer
+	cmd := newVersionCommand(lcc, logger, &stdOut)
+	assert.Equal(t, cmd.Name(), "version")
+}
+
+func TestVersionAction_runAdaptor(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name    string
+		cmd     *cobra.Command
+		args    []string
+		mockSvc func(*mocks.LimaCmdCreator, *mocks.Logger, *gomock.Controller)
+	}{
+		{
+			name: "happy path",
+			cmd: &cobra.Command{
+				Use: "version",
+			},
+			args: []string{},
+			mockSvc: func(lcc *mocks.LimaCmdCreator, logger *mocks.Logger, ctrl *gomock.Controller) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Running"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Running")
+
+				command := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("shell", limaInstanceName, "nerdctl", "version", "--format", "json").Return(command)
+				command.EXPECT().Output()
+			},
+		},
+	}
 
-	require.NoError(t, write.Close())
-	os.Stdout = temp
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
 
-	var buffer bytes.Buffer
-	_, err := io.Copy(&buffer, read)
-	require.NoError(t, err)
-	return buffer.String()
+			ctrl := gomock.NewController(t)
+			logger := mocks.NewLogger(ctrl)
+			lcc := mocks.NewLimaCmdCreator(ctrl)
+			var stdOut bytes.Buffer
+			tc.mockSvc(lcc, logger, ctrl)
+
+			assert.NoError(t, newVersionAction(lcc, logger, &stdOut).runAdapter(tc.cmd, tc.args))
+		})
+	}
 }
 
-func TestVersionCommand(t *testing.T) {
+func TestVersionAction_run(t *testing.T) {
 	t.Parallel()
 
-	cmd := newVersionCommand()
-	assert.Equal(t, cmd.Name(), "version")
-}
+	testCases := []struct {
+		name         string
+		wantErr      error
+		cmd          *cobra.Command
+		mockSvc      func(*mocks.LimaCmdCreator, *mocks.Logger, *gomock.Controller)
+		postRunCheck func(t *testing.T, stdout []byte)
+	}{
+		{
```

Comment:
I think we're still missing the error test case, but everything else looks good in the latest revision

Reply:
Not sure why Linter fails in CI during environment preparation. 

Example:
```
  /usr/bin/tar: ../../../go/pkg/mod/inet.af/tcpproxy@v0.0.0-20220326234310-be3ee21c9fa0/.travis.yml: Cannot open: File exists
  Error: /usr/bin/tar: ../../../go/pkg/mod/inet.af/tcpproxy@v0.0.0-20220326234310-be3ee21c9fa0/tcpproxy.go: Cannot open: File exists
```

Reply:
Yeah the linter error does look really weird, but I think this comment is about 
having a mock where lima.GetVMStatus returns an error, instead of nil, like

```golang
...
    getVMStatusC.EXPECT().Output().Return(nil, errors.new("Some error"))
...
```

Reply:
added in new push